inc/umock_c/umock_c_internal.h

```diff
@@ -1476,6 +1477,14 @@ typedef struct MOCK_CALL_METADATA_TAG
 #define UMOCK_REAL(function_name) \
     MU_C2(umock_real_, function_name)
 
+#ifdef __cplusplus
+#define IGNORED_STRUCT_ARG(struct_type) \
+    { 0 }
+#else
+#define IGNORED_STRUCT_ARG(struct_type) \
+    (struct_type) { 0 }
```

Comment:
>(struct_type) [](start = 4, length = 13)

hm. Why is this needed?

There's also this little gem of a warning: error C4576: a parenthesized type followed by an initializer list is a non-standard explicit type conversion syntax

The following C++ code compiles perfectly... 
```c++
typedef struct T_TAG
{
    int* a;
    int b;
}T;

static void f(T t)
{
    (void)t;
}

int main(void)
{
    f({ 0 });
}
```

asking for a friend... 

Because if in C++ IGNORED_STRUCT_ARG(struct_type)  could expand to just {0}, then it doesn't really need the "struct_type" argument. Next step would be to remove _STRUCT from the macro name. But one is wishful.

Reply:
Yes, unfortunately it does not compile for C.

------

C code that does not compile:

```c
typedef struct HAGA_TAG
{
    int x;
} HAGA;

void uaga(HAGA h)
{
    (void)h;
}

int main(void)
{
    uaga({ 0 });
}
```

--------
C code that compiles:

```c
typedef struct HAGA_TAG
{
    int x;
} HAGA;

void uaga(HAGA h)
{
    (void)h;
}

int main(void)
{
    HAGA haga = { 0 };
    uaga(haga);
}
```

Soooo, myeah. For now I'll leave it as is. I'll try to hunt down someone that can ack whether this is a C compiler issue for MSVC.

---
In reply to: [399338414](https://github.com/Azure/umock-c/pull/154#discussion_r399338414) [](ancestors = 399338414)

Reply:
Turns out that it also fails on gcc, so it seems unlikely that both compilers are wrong, so it seems it is me not seeing it in the standard :(... I guess it stays with (struct_type) {0}.

---
In reply to: [399482131](https://github.com/Azure/umock-c/pull/154#discussion_r399482131) [](ancestors = 399482131,399338414)